network/p2p/inspector/validation/control_message_validation_inspector.go

```diff
@@ -651,9 +664,16 @@ func (c *ControlMsgValidationInspector) inspectRpcPublishMessages(from peer.ID,
 		}
 		c.metrics.OnPublishMessageInspected(errCnt, invalidTopicIdsCount, invalidSubscriptionsCount, invalidSendersCount)
 	}()
+
 	for _, message := range messages[:sampleSize] {
 		if c.networkingType == network.PrivateNetwork {
-			err := c.checkPubsubMessageSender(message)
+			pid, err := peer.IDFromBytes(message.GetFrom())
+			if err != nil {
+				invalidSendersCount++
+				errs = multierror.Append(errs, fmt.Errorf("failed to get peer ID from bytes: %w", err))
+				continue
+			}
+			err = c.checkSenderIdentity(pid)
```

Comment:
what's the cost of running `checkSenderIdentity`? if multiple messages are from the same pid, we would run the `checkSenderIdentity` multiple times, what it make sense to optimize?

Reply:
Good catch we can do inline caching https://github.com/onflow/flow-go/pull/5449/commits/9a4e4fc07c3773b67104b7e39ec31f0176b49e78